src/common/certificate-authorities/inject-system-cas-on-macos.injectable.ts

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import { globalAgent } from "https";
+import execFileInjectable from "../fs/exec-file.injectable";
+import loggerInjectable from "../logger.injectable";
+import { isCertActive } from "./is-cert-active";
+import isMacInjectable from "../vars/is-mac.injectable";
+import { systemCAsInjectorInjectionToken } from "./system-cas-injector-token";
+
+function injectCAs(CAs: string[]) {
+  for (const cert of CAs) {
+    if (Array.isArray(globalAgent.options.ca) && !globalAgent.options.ca.includes(cert)) {
+      globalAgent.options.ca.push(cert);
+    } else {
+      globalAgent.options.ca = [cert];
+    }
+  }
+}
+
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Cheatsheet#other_assertions
+const certSplitPattern = /(?=-----BEGIN\sCERTIFICATE-----)/g;
+
+const injectSystemCAsOnMacOSInjectable = getInjectable({
+  id: "inject-system-cas-on-macOS",
+  instantiate: (di) => {
+    const execFile = di.inject(execFileInjectable);
+    const logger = di.inject(loggerInjectable);
+    const isMac = di.inject(isMacInjectable);
+
+    const execSecurity = async (...args: string[]) => {
+      const output = await execFile("/usr/bin/security", args);
+
+      return output.split(certSplitPattern);
+    };
+
+    if (!isMac) {
+      return async () => {};
+    }
```

Comment:
Musings for future: Hmm. That one idea about being able to have eg. `system-cas.injectable.mac.ts` and `system-cas.injectable.win.ts` instead of code like this could be pretty good.

Reply:
All the yes!!! I was thinking about using folders but that is probably a better idea. Should I do that in this PR?